# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmIonCompile.cpp
# Commit: 28feda14fdec
# Full Hash: 28feda14fdec962ea68c26a0570ea69042a5ad27
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 3
# Description:
#   Bug 1692065 - wasm: Add tagging for JSString* in AnyRef. r=yury
#   
#   Before this commit we would need to box a JSString* in WasmValueBox
#   when we received one from JS and converted to an AnyRef.
#   
# ==============================================================================

diff -r 85605041672d -r 28feda14fdec js/src/wasm/WasmIonCompile.cpp
--- a/js/src/wasm/WasmIonCompile.cpp	Fri Aug 04 14:06:50 2023 +0000
+++ b/js/src/wasm/WasmIonCompile.cpp	Fri Aug 04 14:06:50 2023 +0000
@@ -839,21 +839,21 @@
     curBlock_->setSlot(info().localSlot(slot), def);
   }
 
-  MDefinition* compareIsNull(MDefinition* value, JSOp compareOp) {
+  MDefinition* compareIsNull(MDefinition* ref, JSOp compareOp) {
     MDefinition* nullVal = constantNullRef();
     if (!nullVal) {
       return nullptr;
     }
-    return compare(value, nullVal, compareOp, MCompare::Compare_WasmAnyRef);
-  }
-
-  [[nodiscard]] bool refAsNonNull(MDefinition* value) {
+    return compare(ref, nullVal, compareOp, MCompare::Compare_WasmAnyRef);
+  }
+
+  [[nodiscard]] bool refAsNonNull(MDefinition* ref) {
     if (inDeadCode()) {
       return true;
     }
 
     auto* ins = MWasmTrapIfNull::New(
-        alloc(), value, wasm::Trap::NullPointerDereference, bytecodeOffset());
+        alloc(), ref, wasm::Trap::NullPointerDereference, bytecodeOffset());
 
     curBlock_->add(ins);
     return true;
@@ -6800,12 +6800,12 @@
 
 #ifdef ENABLE_WASM_FUNCTION_REFERENCES
 static bool EmitRefAsNonNull(FunctionCompiler& f) {
-  MDefinition* value;
-  if (!f.iter().readRefAsNonNull(&value)) {
-    return false;
-  }
-
-  return f.refAsNonNull(value);
+  MDefinition* ref;
+  if (!f.iter().readRefAsNonNull(&ref)) {
+    return false;
+  }
+
+  return f.refAsNonNull(ref);
 }
 
 static bool EmitBrOnNull(FunctionCompiler& f) {